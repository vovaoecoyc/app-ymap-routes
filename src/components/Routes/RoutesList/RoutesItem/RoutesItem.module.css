.PointBlock {
  display: flex;
  justify-content: space-between;
  border: 1px solid #d5d5d5;
  padding: 5px;
}

.RemoveIcon {
  font-weight: bold;
  cursor: pointer;
}