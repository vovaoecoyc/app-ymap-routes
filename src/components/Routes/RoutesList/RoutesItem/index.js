import React from 'react';

import { AppContext } from '../../../../Context';
import styles from './RoutesItem.module.css';

class RoutesItem extends React.Component {
  static contextType = AppContext;

  handlerClick = () => {
    this.context.removePoint(this.props.id);
  };

  render() {
    let {
      point: { name },
    } = this.props;
    return (
      <div className={`${styles.PointBlock}`}>
        <span>{name}</span>
        <div onClick={this.handlerClick} className={`${styles.RemoveIcon}`}>
          X
        </div>
      </div>
    );
  }
}

export default RoutesItem;
