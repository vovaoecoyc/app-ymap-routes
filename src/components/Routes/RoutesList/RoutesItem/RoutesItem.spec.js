import React from 'react';
import { shallow, mount } from 'enzyme';

import RoutesItem from '../RoutesItem';
import context from '../../../../MockContext';

describe('RoutesItem container', () => {
  const props = {
    id: 1,
    point: { coordinates: [55.1, 38.1], name: 'Name Point' },
  };

  describe('RoutesItem initial', () => {
    const containerRoutesItem = shallow(<RoutesItem key={1} {...props} />, {
      context: { ...context },
    });

    it('RoutesItem initial snapshot', () => {
      expect(containerRoutesItem).toMatchSnapshot();
    });

    it('Initial render name of point', () => {
      expect(
        containerRoutesItem
          .find('div')
          .find('span')
          .text()
      ).toBe(props.point.name);
    });

    /*it('Remove RoutesItem', () => {
      containerRoutesItem
        .find('div')
        .first()
        .children()
        .find('div')
        .simulate('click');

      expect(containerRoutesItem.context.removePoint(props.id)).toHaveBeenCalledTimes(1);
      containerRoutesItem.unmount();
      // expect(containerRoutesItem.find());
    });*/
  });
});
