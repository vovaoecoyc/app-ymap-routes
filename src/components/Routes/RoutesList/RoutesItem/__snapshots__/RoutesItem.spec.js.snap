// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutesItem container RoutesItem initial RoutesItem initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoutesItem
    id={1}
    point={
      Object {
        "coordinates": Array [
          55.1,
          38.1,
        ],
        "name": "Name Point",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Name Point
        </span>,
        <div
          className="RemoveIcon"
          onClick={[Function]}
        >
          X
        </div>,
      ],
      "className": "PointBlock",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name Point",
        },
        "ref": null,
        "rendered": "Name Point",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "RemoveIcon",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Name Point
          </span>,
          <div
            className="RemoveIcon"
            onClick={[Function]}
          >
            X
          </div>,
        ],
        "className": "PointBlock",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name Point",
          },
          "ref": null,
          "rendered": "Name Point",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "RemoveIcon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "changePointPosition": [Function],
      "getMapInstance": [Function],
      "getMapRef": [Function],
      "getPoints": [Function],
      "loadPoints": [Function],
      "mapInstance": null,
      "mapRef": null,
      "points": Array [
        Object {
          "coordinates": Array [
            55.1,
            38.1,
          ],
          "name": "Name Point1",
        },
        Object {
          "coordinates": Array [
            55.9,
            38,
          ],
          "name": "Name Point2",
        },
      ],
      "removePoint": [Function],
      "setMapInstance": [Function],
      "setMapRef": [Function],
    },
  },
}
`;
